packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,43 +7,84 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
+import {TDaysSelected} from './types';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	daysSelected: TDaysSelected;
 	disabled?: boolean;
-	onClick: (date: Date) => void;
+	range?: boolean;
+	onClick: (date: Date, range?: boolean) => void;
+}
+
+interface IInterval {
+	start: Date;
+	end: Date;
+}
+
+function isWithinInterval(dirtyDate: Date, dirtyInterval: IInterval) {
+	const interval = dirtyInterval || {};
+	const time = dirtyDate.getTime();
+	const startTime = interval.start.getTime();
+	const endTime = interval.end.getTime();
```

Comment:
I can't see any differences when removing `|| {}`. In this case, should I warn the user about the missing `end|start` properties? I can't see defining a default value for it like new Date(). It may broke or introduce some weird usages(like ppl considering when not passing a start or an end the new Date() will be considered.

Reply:
> should I warn the user about the missing `end|start` properties?

In general when it comes to warnings, it's a trade-off. 🤷 

- If you warn, it might be spammy, but also possibly helpful.
- And irrespective of whether you warn, you should degrade "gracefully" in some way (requires you to come up with a definition of what "degrade gracefully" means, but avoids the spam, at the cost of potential eliding helpful information).

In this specific instance, they'd only be missing because of _you_ didn't provide them in your fallback object (`{}`), so I don't think I'd want to warn about that. In any case, I only see one caller of `isWithinInterval`, and it always passes an object there, so it doesn't look like the fallback would ever be used. So I'd say, just delete the `|| {}` (in fact, I am almost surprised that TS doesn't warn that its a useless expression). That still leaves open the broader question about prop validation, but that would/should occur "upstream" of here, in the component. Getting rid of the fallback removes the obvious local inconsistency internal to this function.

Reply:
> In general when it comes to warnings, it's a trade-off. 🤷

> If you warn, it might be spammy, but also possibly helpful.
> And irrespective of whether you warn, you should degrade "gracefully" in some way (requires you to come up with a definition of what "degrade gracefully" means, but avoids the spam, at the cost of potential eliding helpful information).

Thanks for the explanation! 

Considering the current usage I understood the fallback may be unnecessary. So, I'll remove `|| {}` :)